Source/drlg_l2.cpp

```diff
@@ -1751,16 +1747,13 @@ static void DRLG_L2PlaceRndSet(const BYTE *miniset, int rndper)
 
 static void DRLG_L2Subs()
 {
-	int x, y, i, j, k, rv;
-	BYTE c;
-
-	for (y = 0; y < DMAXY; y++) {
-		for (x = 0; x < DMAXX; x++) {
+	for (int y = 0; y < DMAXY; y++) {
+		for (int x = 0; x < DMAXX; x++) {
 			if ((x < nSx1 || x > nSx2) && (y < nSy1 || y > nSy2) && GenerateRnd(4) == 0) {
-				c = BTYPESL2[dungeon[x][y]];
+				int8_t c = BTYPESL2[dungeon[x][y]];

```

Comment:
Should this be uint8_t?

```suggestion
				uint8_t c = BTYPESL2[dungeon[x][y]];
```